### PR TITLE
Changed IPart#onNeighborChanged() to add vanilla arguments

### PR DESCRIPTION
**Pull request by yueh**
Allows more precise checks to limit an update to the facing side.

### CODE REVIEW

File: src/main/java/appeng/parts/CableBusContainer.java [hunk 1]
```diff
@@ -832,7 +832,7 @@ public void onNeighborChanged()
 			final IPart part = this.getPart( s );
 			if( part != null )
 			{
-				part.onNeighborChanged();
+				part.onNeighborChanged(w, pos, neighbor);
```
Comment (orod-org) [inline src/main/java/appeng/parts/CableBusContainer.java]: MINOR '(' is not followed by whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
MINOR ')' is not preceded with whitespace. [rule](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)

**Comment by orod-org** [comment]
SonarQube analysis reported 33 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 18 major
* ![MINOR][MINOR] 11 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PartP2PFEPower.java#L100](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L100): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PFEPower.java#L115](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L115): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PRFPower.java#L98](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L98): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PRFPower.java#L113](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L113): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [AEBasePart.java#L499](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/AEBasePart.java#L499): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [AEBasePart.java#L504](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/AEBasePart.java#L504): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L102](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L102): A "NullPointerException" could be thrown; "getOutput" is nullable here. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L117](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L117): A "NullPointerException" could be thrown; "getOutput" is nullable here. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L130](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L130): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L131](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/1a96ba7e67f4ee3ef80e91ce8cb131ea32dfbf25/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L131): Either remove or fill this block of code. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00108)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by orod-org** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 4 critical
* ![MAJOR][MAJOR] 18 major
* ![MINOR][MINOR] 9 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [PartP2PFEPower.java#L100](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L100): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PFEPower.java#L115](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L115): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PRFPower.java#L98](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L98): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![CRITICAL][CRITICAL] [PartP2PRFPower.java#L113](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L113): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
1. ![MAJOR][MAJOR] [AEBasePart.java#L499](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/AEBasePart.java#L499): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [AEBasePart.java#L504](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/AEBasePart.java#L504): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L102](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L102): A "NullPointerException" could be thrown; "getOutput" is nullable here. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L117](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L117): A "NullPointerException" could be thrown; "getOutput" is nullable here. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L130](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L130): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L131](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L131): Either remove or fill this block of code. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00108)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by yueh** [comment]
test this please.

**Comment by orod-org** [comment]
SonarQube analysis reported 14 issues
* ![MAJOR][MAJOR] 10 major
* ![MINOR][MINOR] 4 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AEBasePart.java#L499](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/AEBasePart.java#L499): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [AEBasePart.java#L504](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/AEBasePart.java#L504): Remove this unused method parameter "player". [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L120](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L120): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L121](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L121): Either remove or fill this block of code. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L184](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L184): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PFEPower.java#L209](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PFEPower.java#L209): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PRFPower.java#L118](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L118): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PRFPower.java#L119](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L119): Either remove or fill this block of code. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS00108)
1. ![MAJOR][MAJOR] [PartP2PRFPower.java#L168](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L168): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [PartP2PRFPower.java#L193](https://github.com/AppliedEnergistics/Applied-Energistics-2/blob/d17754b98cea9391dd78e1455ed8686c0d771cee/src/main/java/appeng/parts/p2p/PartP2PRFPower.java#L193): Either log or rethrow this exception. [rule](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1166)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**